contracts/exchange/test/transactions_unit_tests.ts

```diff
@@ -0,0 +1,694 @@
+                    [transaction],
+                    [validSignature],
+                ),
+            );
+
+            // Ensure that the correct number of events were logged.
+            const logs = receipt.logs as Array<LogWithDecodedArgs<TestTransactionsTransactionExecutionEventArgs>>;
+            expect(logs.length).to.be.eq(2);
+
+            // Ensure that the correct events were logged.
+            expect(logs[0].event).to.be.eq('ExecutableCalled');
+            expect(logs[0].args.data).to.be.eq(constants.NULL_BYTES);
+            expect(logs[0].args.returnData).to.be.eq('0xdeadbeef');
+            expect(logs[1].event).to.be.eq('TransactionExecution');
+            expect(logs[1].args.transactionHash).to.eq(transactionHash);
+        });
+
+        it('should succeed if the both calls to executeTransaction succeed', async () => {
+            const currentTimestamp = await getLatestBlockTimestampAsync();
+
+            // Create two transactions that will succeed when used to call `batchExecuteTransactions()`.
+            const transaction1 = {
+                ...EMPTY_ZERO_EX_TRANSACTION,
+                expirationTimeSeconds: new BigNumber(currentTimestamp).plus(10),
+                signerAddress: accounts[0], // This is different than the account that will be used to send.
+                data: getExecutableCallData(true, constants.NULL_BYTES, '0xdeadbeef'),
+                domain,
+            };
+            const transaction2 = {
+                ...EMPTY_ZERO_EX_TRANSACTION,
+                expirationTimeSeconds: new BigNumber(currentTimestamp).plus(10),
+                signerAddress: accounts[1], // Different than transaction1's signer address
+                data: getExecutableCallData(true, constants.NULL_BYTES, '0xbeefdead'),
+                domain,
+            };
+            const transactionHash1 = transactionHashUtils.getTransactionHashHex(transaction1);
+            const transactionHash2 = transactionHashUtils.getTransactionHashHex(transaction2);
+
+            // Verify that the returndata of the transaction is 0xDEADBEEF
+            const result = await transactionsContract.batchExecuteTransactions.callAsync(
+                [transaction1, transaction2],
+                [randomSignature(), randomSignature()],
+                {
+                    from: accounts[0],
+                },
+            );
+
+            // Create an abiEncoder for bytes. This will be used to decode the result and encode what
+            // is expected.
+            const abiEncoder = AbiEncoder.create('bytes');
+
+            // Ensure that the result contains the abi-encoded bytes "0xdeadbeef"
+            const encodedDeadbeef = abiEncoder.encode('0xdeadbeef');
+            expect(result.length).to.be.eq(2);
+            expect(
+                result[0] ===
+                    '0x0000000000000000000000000000000000000000000000000000000000000020'.concat(
```

Comment:
I agree. My rationale for not doing this is that a fix to the AbiEncoder will actually allow us to use that for this check. At the moment, I have to do this because `AbiEncoder.create('bytes').decode(result[0])` will not correctly decode the bytes (it will actually just return the 32 byte slot containing the length). I've added an Asana backlog task for this, so it should actually be completed at some point. What do you think of this?

Reply:
If it's a hassle, nbd.